usr/sbin/regionsrv-enabler-azure

```diff
@@ -78,16 +97,18 @@ utils.start_logging()
 service_name = 'guestregister'
 license_type = get_license_type()
 
-if 'BYOS' in license_type and not utils.uses_rmt_as_scc_proxy():
+if ('BYOS' in license_type and has_license_changed(license_type) and
+    not utils.uses_rmt_as_scc_proxy()):
     run_command(['registercloudguest', '--clean'])
     run_command(['systemctl', 'disable', service_name])
 elif license_type and 'BYOS' not in license_type:
     # if no license type, system could be a BYOS without AHB
     # or PAYG without AHB
     update_server = utils.get_smt()
     if update_server:
-        if utils.is_registered(update_server.get_FQDN()):
+        if not has_license_changed(license_type) and utils.is_registered(update_server.get_FQDN()):
```

Comment:
This line pretty long compared to the rest of the code.

Reply:
Fixed